File: LeetCode-Cpp/KidsWiththeGreatestNumberofCandies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        int maxCandies = 0;
        for (int candy : candies) {
            if (candy > maxCandies) {
                maxCandies = candy;
            }
        }

        std::vector<bool> result;
        for (int candy : candies) {
            if (candy + extraCandies >= maxCandies) {
                result.push_back(true);
            } else {
                result.push_back(false);
            }
        }

        return result;
        }
};

File: LeetCode-Python/KidsWiththeGreatestNumberofCandies.py
lass Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
      max_candies = max(candies)  # En fazla şeker sayısını bul
      result = []

      for kid_candies in candies:
          if kid_candies + extraCandies >= max_candies:
              result.append(True)
          else:
              result.append(False)

      return result
